### PR TITLE
Fix bug with estimation years

### PR DESCRIPTION
**Pull request by zmbc**
### Description
- *Category*: bugfix
- *JIRA issue*: None

### Changes and notes

Corrected validation of the `years` parameter.

### Testing

Have not tested this.

### CODE REVIEW

**Review by albrja — APPROVED** [review]

File: src/vivarium_inputs/extract.py [hunk 1]
```diff
@@ -104,14 +104,19 @@ def extract_data(
...
-    if years == None:  # default to most recent year
+    if years is None:  # default to most recent year
         year_id = gbd.get_most_recent_year()
     elif years == "all":
         year_id = None
     else:
         estimation_years = gbd.get_estimation_years()
-        if years not in estimation_years:
-            raise ValueError(f"years must be in {estimation_years}. You provided {years}.")
+        if not isinstance(years, list):
```
Comment (rmudambi) [inline src/vivarium_inputs/extract.py:113]: I think this is slightly cleaner
```python
else:
    year_id = years if isinstance(years, list) else [years]
    not_estimated_years = set(year_id).difference(gbd.get_estimation_years())
    if not_estimated_years:
        raise ...
```

File: src/vivarium_inputs/utility_data.py [hunk 2]
```diff
@@ -69,12 +69,19 @@ def get_demographic_dimensions(
...
-    if years == "all":
+    if years is None:
+        years = [gbd.get_most_recent_year()]
+    elif years == "all":
         years = range(min(estimation_years), max(estimation_years) + 1)
-    else:
-        if years and years not in estimation_years:
-            raise ValueError(f"years must be in {estimation_years}. You provided {years}.")
-        years = [years] if years else [gbd.get_most_recent_year()]
+    elif not isinstance(years, list):
```
Comment (rmudambi) [inline src/vivarium_inputs/utility_data.py:76]: Why are the solutions different in these two files? The problem appears to be the same.

Comment (zmbc) [reply src/vivarium_inputs/utility_data.py]: The solutions are slightly different; one is calling GBD shared functions and the other is creating a data structure from scratch (years = None) won't work in the latter case. But they were more different than they needed to be; I've updated this.

**Review by rmudambi — APPROVED** [review]

**Comment by zmbc** [comment]
Superseded by #351.

**Pull request closed without merging**